app/javascript/packs/leafletMap.js

```diff
@@ -71,14 +71,45 @@ export async function setupMap() {
   map.on("click", onClick);
 }
 
-function addPin(e, pinNumber) {
-  pins[pinNumber] = L.marker(e.latlng, { icon: pinIcons[pinNumber] });
+function formatQueryParameters(position){
+  return "?long="+position.lng+"&lat="+position.lat
```

Comment:
We want to keep the current query params like `locale="en"`.

```suggestion
    const query = new URLSearchParams(window.location.search);
    query.set("long", position.lng);
    query.set("lat", position.lat);
    return query.toString();
```

(Not tested!)